src/sInv.sol

```diff
@@ -225,7 +239,7 @@ contract sINV is ERC4626 {
      * @dev Sets the new revenue accrual and K updating period.
      * @param _period The new revenue and K updating period.
      */
-    function setPeriod(uint _period) external onlyGov {
+    function setPeriod(uint256 _period) external onlyGov {
```

Comment:
issue:

when setting this to a higher value...
1. you have risk of double-counting the period revenue [here](https://github.com/InverseFinance/sINV/blob/72021a826222f8f3999a82c9de6033bbeb21a7b0/src/sInv.sol#L168) in scenario where `lastBuy` was originally in a period that was already released, but is now in the current period.
2. a temporary loss is experienced (perhaps not too important).


when setting this to a lower value...
1. profit might be lost. E.g. in a scenario where period is set from 1 week --> 1 day, a buy from 6 days ago would never get released to stakers.
2. it can be sandwiched to steal the difference between profit released in old setting vs new setting.

Reply:
further, modifying this setting will impact the `k` value and therefore may be backrunnable as it disrupts the buy price differently than the inteded linearly decay/growth.
see [here](https://github.com/InverseFinance/sINV/blob/72021a826222f8f3999a82c9de6033bbeb21a7b0/src/sInv.sol#L183).

Reply:
I don't think profit is ever lost, as revenue can only be locked up for a maximum of two periods.

In the case of case of extending the period you definitely run into an issue of revenue being locked for longer than would have been expected, but I don't think it'll be double counted. Either the `updatePeriodRevenue(...)` function is called before the longer period is set, and the revenue is released as profit, or it's called after the longer period is set, and the locked revenue will follow the release time of the new period.

Will run a test to make sure.

Reply:
In the case of setting a shorter period, there could definitely be some fuckery with an attacker sandwiching the call.